src/main/java/uk/gov/hmcts/reform/civil/model/dq/HomeTypeValidator.java

```diff
@@ -0,0 +1,25 @@
+package uk.gov.hmcts.reform.civil.model.dq;
+
+import org.apache.commons.lang3.StringUtils;
+import uk.gov.hmcts.reform.civil.enums.HomeTypeOptionSpec;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+public class HomeTypeValidator implements ConstraintValidator<ValidHomeType, HomeDetails> {
+
+    /**
+     * Checks if HomeDetails type fields are valid.
+     *
+     * @param value   mandatory
+     * @param context unused
+     * @return true if and only if value.type is not null, and it is either different from OTHER
+     *     or value.typeOtherDetails is not blank
+     */
+    @Override
+    public boolean isValid(HomeDetails value, ConstraintValidatorContext context) {
```

Comment:
Not sure I understand the need for validation here. The CCD UI handles the empty text field validation out of the box, so my view is we cannot test it from frontend.